### PR TITLE
Use Dates again instead of Unix timestamps in milliseconds in auth lib code

### PR DESCRIPTION
**Pull request by arsalansufi**
## Overview

Since @jonahkagan identified a fix for https://github.com/votingworks/vxsuite/issues/3233 :tada:, we can use Dates again instead of Unix timestamps in milliseconds in auth lib code. (The former are just a bit more convenient than the latter.) This PR switches back to Dates accordingly.

## Testing

- [x] Updated automated tests
- [x] Tested session time limits manually

## Checklist

- [ ] ~I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced~ N/A
- [ ] ~I have added JSDoc comments to any newly introduced exports~ N/A

### CODE REVIEW

File: apps/mark/backend/src/app.test.ts [hunk 1]
```diff
@@ -68,57 +68,6 @@ test('uses default machine config if not set', async () => {
   });
 });
 
-test('auth', async () => {
```
Comment (arsalansufi) [inline apps/mark/backend/src/app.test.ts:71]: I actually previously moved these tests to [app.auth.test.ts](https://github.com/votingworks/vxsuite/blob/main/apps/mark/backend/src/app.auth.test.ts) and forgot to remove the old ones still in this file

File: apps/scan/backend/src/scanners/custom/app_config.test.ts [hunk 2]
```diff
@@ -295,7 +295,7 @@ test('export CVRs to USB in deprecated VotingWorks format', async () => {
   );
 });
 
-test('setPrecinctSelection will reset polls to closed and update auth instance', async () => {
```
Comment (arsalansufi) [inline apps/scan/backend/src/scanners/custom/app_config.test.ts:298]: A relic of a much earlier iteration of auth code

**Review by eventualbuddha — APPROVED** [review]

File: apps/scan/backend/src/scanners/custom/app_config.test.ts (hunk 2 above)
Comment (eventualbuddha) [reply apps/scan/backend/src/scanners/custom/app_config.test.ts:298]: Untitled

**Review by eventualbuddha — APPROVED** [review]
`DateTime` changes look great. Thanks!

**Pull request merged**